src/ParticleDA.jl

```diff
@@ -78,7 +78,7 @@ the type of `model_data`.
 function sample_initial_state! end
 
 """
-    ParticleDA.update_state_deterministic!(state, model_data)
+    ParticleDA.update_state_deterministic!(state, model_data, ip)
```

Comment:
What is the specific need for having access to the particle index in `update_state_deterministic!` in SPEEDY (note that the particle index will in general not be the same as `threadid`)? For the particle filter to be correct the state updates shouldn't depend on the particle index. If this is needed to access a thread specific buffer or similar then you can use `threadid` within `update_state_deterministic!`

Reply:
Thank you for the suggestion, the threadid is a much better solution. The threadid is needed in the `update_state_deterministic!` in Speedy as individual particles have separate sub folders with the naming structure of rank_id/threadid.